core/nodejsActionBase/runner.js

```diff
@@ -169,4 +169,16 @@ function assertMainIsFunction(userScriptMain, main) {
     }
 }
 
+/**
+ * Evals the code to execute. This is a global function so that the eval is in the global context
+ * and hence functions which use variables without 'var' are permitted.
+ */
+function evalScript(main, whatToRequire, code) {
```

Comment:
Might be more concise to move `whatToRequire`to the end of the parameter list and call `evalScript`with three parameters in the first case and with two  parameters in the `plain old JS case`. Your `if`statement would work as expected still.

Other than that the change does look fine.